tests/optimize/test_backtesting_adjust_position.py

```diff
@@ -197,3 +197,145 @@ def test_backtest_position_adjustment_detailed(default_conf, fee, mocker, levera
+
+
+def test_backtest_exit_trade_with_open_entry_order(default_conf, fee, mocker,) -> None:
+    default_conf['use_exit_signal'] = False
+    default_conf['max_open_trades'] = 10
+    default_conf['stake_amount'] = 8
+    mocker.patch(f'{EXMS}.get_fee', fee)
+
+    backtesting = Backtesting(default_conf)
+    backtesting._set_strategy(backtesting.strategylist[0])
+    pair = 'XRP/USDT:USDT'
+    row = [
+        pd.Timestamp(year=2023, month=1, day=1, hour=5, minute=0),
+        2.1,  # Open
+        2.2,  # High
+        1.9,  # Low
+        2.1,  # Close
+        1,  # enter_long
+        0,  # exit_long
+        0,  # enter_short
+        0,  # exit_short
+        '',  # enter_tag
+        '',  # exit_tag
+        ]
+
+    backtesting.strategy.leverage = MagicMock(return_value=10.0)
+    trade = backtesting._enter_trade(pair, row=row, direction='long')
+    assert trade
+
+    print(f'trade.liquidation_price: {trade.liquidation_price}')
+
+    # position adjust creating exit order
+    def custom_entry_price(proposed_rate, **kwargs):
+        return proposed_rate * 0.95
+    
+    backtesting.strategy.custom_entry_price = custom_entry_price
+    backtesting.strategy.adjust_trade_position = MagicMock(return_value=5)
+
+    # wait for stoploss trigger event
+    row_sl_tgrd = [
+        pd.Timestamp(year=2023, month=1, day=1, hour=5, minute=30),
+        1.1,  # Open
+        1.2,  # High
+        0.9,  # Low
+        1.11,  # Close
+        0,  # enter_long
+        0,  # exit_long
+        0,  # enter_short
+        0,  # exit_short
+        '',  # enter_tag
+        '',  # exit_tag
+        ]
+    res = backtesting._check_trade_exit(trade, row_sl_tgrd, row_sl_tgrd[0].to_pydatetime())
+
+    # TODO count and validate that there is a open entry order
+    print(res)
+    print(f'exit reason : {res.exit_reason}')
+
+    # TODO This should FAIL curently as exits is not triggered if trade.has_open_orders
+    assert res
+    assert res.exit_reason == ExitType.STOP_LOSS.value
+
+def test_backtest_liquidation_with_open_exit_order(default_conf_usdt, fee, mocker,) -> None:
+    default_conf_usdt['use_exit_signal'] = False
+    mocker.patch(f'{EXMS}.get_fee', fee)
+    mocker.patch(f"{EXMS}.get_min_pair_stake_amount", return_value=0.00001)
+    mocker.patch(f"{EXMS}.get_max_pair_stake_amount", return_value=float('inf'))
+    mocker.patch(f"{EXMS}.get_max_leverage", return_value=100)
+    mocker.patch("freqtrade.optimize.backtesting.price_to_precision", lambda p, *args: p)
+    patch_exchange(mocker)
+
+    default_conf_usdt['max_open_trades'] = 10
+    default_conf_usdt['stake_amount'] = 10
+    default_conf_usdt['trading_mode'] = 'futures'
+    default_conf_usdt['margin_mode'] = 'isolated'
+    default_conf_usdt['candle_type_def'] = CandleType.FUTURES
+    default_conf_usdt['stake_currency'] = 'USDT'
+    default_conf_usdt['exchange']['pair_whitelist'] = ['XRP/USDT:USDT']
```

Comment:
I'm getting a "FAILED tests/optimize/test_backtesting_adjust_position.py::test_backtest_liquidation_with_open_exit_order - freqtrade.exceptions.OperationalException: No pair in whitelist." error , any idea why?

Reply:
Don't think there's a markets entry for XRP/USDT:USDT in the default markets mock ... making it an invalid pair